### PR TITLE
sqlstats: add tracer that prints executed SQL queries

### PR DESCRIPTION
**Pull request by crawshaw**

### CODE REVIEW

File: sqlstats/sqlstats.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+func (t *Tracer) Query(prepCtx context.Context, id sqlite.TraceConnID, query string, duration time.Duration, err error) {
+	stats := t.queryStats(query)
+
+	atomic.AddInt64(&stats.count, 1)
+	atomic.AddInt64(&stats.duration, int64(duration))
+	if err != nil {
+		atomic.AddInt64(&stats.errors, 1)
+	}
+}
```
Comment (bradfitz) [inline sqlstats/sqlstats.go:83]: so this part runs on each query? a sync.Map for `t.queries` is probably a reasonable use of sync.Map considering that once it's seen each unique query it's never written to

Comment (crawshaw) [reply sqlstats/sqlstats.go:83]: Note that only the first time a query is seen is the write mutex held. After that, only the read lock is needed. Given that means ~never holding the write lock in steady state I think this works out more efficient than a sync.Map.

Comment (bradfitz) [reply sqlstats/sqlstats.go:83]: A sync.Map fast path is an atomic load. This PR's fast path is two atomic adds.

A sync.Map would be faster.

**Review by bradfitz — APPROVED** [review]

File: sqlstats/sqlstats.go (hunk 1 above)
Comment (crawshaw) [reply sqlstats/sqlstats.go:89]: Added a TODO, as I'm trying to clean my current open PRs, will come back to it.

**Pull request merged**